xmtp_mls/src/identity/xmtp_id/identity.rs

```diff
@@ -71,112 +77,127 @@ impl Identity {
+                    sign_with_legacy_key(
+                        signature_request.signature_text(),
+                        legacy_signed_private_key,
+                    )
+                    .await?,
+                ))
+                .await?;
+            let identity_update = signature_request.build_identity_update()?;
+            api_client.publish_identity_update(identity_update).await?;
 
-        // We can pre-sign the request with an installation key signature, since we have access to the key
-        signature_request
-            .add_signature(Box::new(
-                sign_with_installation_key(
-                    signature_request.signature_text(),
-                    sized_installation_key(signature_keys.private())?,
-                )
-                .await?,
-            ))
-            .await?;
+            let identity = Self {
+                inbox_id: inbox_id.clone(),
+                installation_keys: signature_keys,
+                credential: create_credential(inbox_id)?,
+                signature_request: None,
+            };
 
-        signature_request
-            .add_signature(Box::new(
-                sign_with_legacy_key(
-                    signature_request.signature_text(),
-                    legacy_signed_private_key,
-                )
-                .await?,
-            ))
-            .await?;
+            Ok(identity)
+        } else {
+            let nonce = rand::random::<u64>();
+            let inbox_id = generate_inbox_id(&address, &nonce);
+            let mut builder = SignatureRequestBuilder::new(inbox_id.clone());
+            builder = builder.create_inbox(member_identifier.clone(), nonce);
 
-        let identity_update = signature_request.build_identity_update()?;
+            let mut signature_request = builder
+                .add_association(installation_public_key.to_vec().into(), member_identifier)
+                .build();
 
-        api_client.publish_identity_update(identity_update).await?;
+            // We can pre-sign the request with an installation key signature, since we have access to the key
+            signature_request
+                .add_signature(Box::new(
+                    sign_with_installation_key(
+                        signature_request.signature_text(),
+                        sized_installation_key(signature_keys.private())?,
+                    )
+                    .await?,
+                ))
+                .await?;
 
-        let identity = Self {
-            inbox_id: inbox_id.clone(),
-            installation_keys: signature_keys,
-            credential: create_credential(inbox_id)?,
-            signature_request: None,
-        };
+            let identity = Self {
+                inbox_id: inbox_id.clone(),
+                installation_keys: signature_keys,
+                credential: create_credential(inbox_id.clone())?,
+                signature_request: Some(signature_request),
```

Comment:
Can we get rid of the Option<SignatureRequest>?

Or, when would we have a None here

Also, when this signature request is resolved will we:

1. Update this identity struct with the `Resolved` signature 
2. resolve to a new Identity type like `SignedIdentity` or otherwise


Not sure which one i would prefer. with 1. we would have to probably have todeal with checking `signed.is_some()` if an `Identity` is signed, but with `SignedIdentity` we would have the added complexity of another struct

Maybe there's a better design too?

Reply:
> Or, when would we have a None here
After SDK created an `Identity` instance, it might ask user wallet to sign the signature_request. After signing, this will be None. It's similar to 1.

Reply:
Also in the case that you are using the SDK for the second+ time (have already done the signing) the `signature_request` will always be None